examples/todo/src/openapi-spec.ts

```diff
@@ -0,0 +1,59 @@
+// Copyright IBM Corp. 2018,2020. All Rights Reserved.
+// Node module: @loopback/example-todo
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {ApplicationConfig} from '@loopback/core';
+import fs from 'fs';
+import {dump} from 'js-yaml';
+import {TodoListApplication} from './application';
+
+/**
+ * Save the OpenAPI spec to the given file
+ * @param outFile - File name for the spec
+ * @param options - Application config
+ */
+export async function dumpOpenApiSpec(
+  outFile: string,
+  options?: ApplicationConfig,
+) {
+  options = options ?? {};
+  options.rest = {...options.rest, listenOnStart: false};
+  const app = new TodoListApplication(options);
+  await app.boot();
+
+  const spec = await app.restServer.getApiSpec();
+  if (outFile === '-' || outFile === '') {
+    const json = JSON.stringify(spec, null, 2);
+    console.log('%s', json);
+    return spec;
+  }
+  const fileName = outFile.toLowerCase();
+  if (fileName.endsWith('.yaml') || fileName.endsWith('.yml')) {
+    const yaml = dump(spec);
+    fs.writeFileSync(outFile, yaml, 'utf-8');
+  } else {
+    const json = JSON.stringify(spec, null, 2);
+    fs.writeFileSync(outFile, json, 'utf-8');
+  }
+  console.log('The OpenAPI spec is saved to %s.', outFile);
+  return spec;
```

Comment:
I am concerned about the amount of code we are generating here. First, it makes the `open-api.ts` script more difficult to understand. Secondly, and this is more important, if we discover a bug in this code, then we won't be able to distribute the fix to our users.

I am proposing to move the implementation of this function to `RestApplication` and/or `RestServer` class, so that the scaffolded function can be shortened to something like this:

```ts
const app = new TodoListApplication(options);
await app.boot();
await app.exportOpenApiSpec(outFile);
```